src/common/src/main/scala/osmesa/common/functions/osm/package.scala

```diff
@@ -499,93 +506,115 @@ package object osm {
     }
   }
 
-  private def connectSegments(segments: GenTraversable[Line])(implicit geometryFactory: GeometryFactory): GenTraversable[Line] =
-    connectSegments(segments.map(_.jtsGeom.getCoordinateSequence).map(s => new VirtualCoordinateSequence(Seq(s))))
+  private def connectSegments(segments: GenTraversable[jts.Geometry])(implicit geometryFactory: GeometryFactory): GenTraversable[LineString] =
+    segments
+      .map({ geom => if (classTag[jts.LineString].runtimeClass.isInstance(geom)) Some(geom.asInstanceOf[jts.LineString]) else None })
```

Comment:
What's the story behind this `instanceof` check? Is it equivalent to `geom.isInstanceOf[jts.LineString]`?

Reply:
It will probably be the same here, but the use of `classTag` avoids type erasure